src/flask_odoo/__init__.py

```diff
@@ -1,5 +1,159 @@
+        common = xmlrpc.client.ServerProxy(f"{ODOO_URL}/xmlrpc/2/common")
+        return common
+
+    @property
+    def common(self):
+        ctx = _app_ctx_stack.top
+        if ctx is not None:
+            if not hasattr(ctx, "odoo_common"):
+                ctx.odoo_common = self.create_common_proxy()
+            return ctx.odoo_common
+
+    def authenticate(self):
+        """Returns a user identifier (uid) used in authenticated calls."""
+        ODOO_DB = current_app.config["ODOO_DB"]
+        ODOO_USERNAME = current_app.config["ODOO_USERNAME"]
+        ODOO_PASSWORD = current_app.config["ODOO_PASSWORD"]
+        uid = self.common.authenticate(
+            ODOO_DB, ODOO_USERNAME, ODOO_PASSWORD, {}
+        )
+        return uid
+
+    @property
+    def uid(self):
+        ctx = _app_ctx_stack.top
+        if ctx is not None:
+            if not hasattr(ctx, "odoo_uid"):
+                ctx.odoo_uid = self.authenticate()
+            return ctx.odoo_uid
+
+    def create_object_proxy(self):
+        ODOO_URL = current_app.config["ODOO_URL"]
+        object = xmlrpc.client.ServerProxy(f"{ODOO_URL}/xmlrpc/2/object")
+        return object
+
+    @property
+    def object(self):
+        ctx = _app_ctx_stack.top
+        if ctx is not None:
+            if not hasattr(ctx, "odoo_object"):
+                ctx.odoo_object = self.create_object_proxy()
+            return ctx.odoo_object
+
+    def __getitem__(self, key):
+        ODOO_DB = current_app.config["ODOO_DB"]
+        ODOO_PASSWORD = current_app.config["ODOO_PASSWORD"]
+        return Model(key, self.object, ODOO_DB, self.uid, ODOO_PASSWORD)
+
+
+class Model:
+    """Simplifies working with Odoo models through XML RPC.
+
+    Args:
+        name: Odoo model name.
+        proxy: XML RPC server proxy.
+        db: The database to use.
+        uid: The user id (retrieved through `authenticate`).
+        password: The user’s password.
+
+    Examples:
+        >>> odoo["res.partner"].execute_kw("check_access_rights", "read",
+        ...                                raise_exception=False)
+        true
+
+    """
+
+    def __init__(
+        self,
+        name: str,
+        proxy: xmlrpc.client.ServerProxy,
+        db: str,
+        uid: int,
+        password: str,
+    ):
+        self.name = name
+        self.proxy = proxy
+        self.db = db
+        self.uid = uid
+        self.password = password
+
+    def execute_kw(self, method_name, *args, **kwargs):
+        """Call `method_name` on this model and return the result."""
+        return self.proxy.execute_kw(
+            self.db,
+            self.uid,
+            self.password,
+            self.name,
+            method_name,
+            args,
+            kwargs,
+        )
+
+    def __getattr__(self, name):
+        return Method(name, self)
+
+    def __repr__(self):
+        return (
+            f"<Model(name='{self.name}', proxy={self.proxy}, db='{self.db}')>"
+        )
+
+
+class Method:
```

Comment:
I love these classes, this is some python magic right here.

Reply:
The `__xxx__` methods are actually referred to as "magic" methods 😉